diff --git a/meson.build b/meson.build
index 41002a5..204bee2 100644
--- a/meson.build
+++ b/meson.build
@@ -500,6 +500,39 @@ if feature_conf.get('CAIRO_HAS_GL_SURFACE', 0) == 0 and feature_conf.get('CAIRO_
   endif
 endif
 
+egl_lib_prefix = ''
+if feature_conf.get('CAIRO_HAS_GL_SURFACE', 0) == 0 and feature_conf.get('CAIRO_HAS_GLESV2_SURFACE', 0) == 0 and \
+   feature_conf.get('CAIRO_HAS_GLESV3_SURFACE', 0) == 0 and ['auto', 'angle'].contains(gl_backend)
+  glesv2_dep = cc.find_library('libGLESv2', required: gl_backend == 'angle')
+  if glesv2_dep.found()
+    if cc.has_header('GLES3/gl3.h', required: false, dependencies: glesv2_dep) and cc.has_header('GLES3/gl3ext.h', required: false, dependencies: glesv2_dep)
+      deps += [glesv2_dep]
+      need_egl_functions = true
+      egl_lib_prefix = 'lib'
+      feature_conf.set('CAIRO_HAS_GLESV3_SURFACE', 1)
+      built_features += [{
+        'name': 'cairo-glesv3',
+        'source-key': 'cairo-gl',
+        'description': 'OpenGLESv3 surface backend',
+        'deps': [glesv2_dep],
+      }]
+    else
+      if cc.has_header('GLES2/gl2.h', required: true,dependencies: glesv2_dep) and cc.has_header('GLES2/gl2ext.h', required: true, dependencies: glesv2_dep)
+        deps += [glesv2_dep]
+        need_egl_functions = true
+        egl_lib_prefix = 'lib'
+        feature_conf.set('CAIRO_HAS_GLESV2_SURFACE', 1)
+        built_features += [{
+          'name': 'cairo-glesv2',
+          'source-key': 'cairo-gl',
+          'description': 'OpenGLESv2 surface backend',
+          'deps':[glesv2_dep],
+        }]
+      endif
+    endif
+  endif
+endif
+
 cogl_dep = dependency('cogl-2.0-experimental',
   version: '>= 1.17.2',
   required: get_option('cogl'),
@@ -555,7 +588,7 @@ if need_egl_functions
         egl_extra_deps += csi_dep
       endif
 
-      foreach libname : ['EGL', 'egl13', 'egl12', 'egl11']
+      foreach libname : [egl_lib_prefix + 'EGL', 'egl13', 'egl12', 'egl11']
         dep = cc.find_library(libname, required: false)
         if dep.found() and cc.has_function('eglGetError', dependencies: [dep])
           egl_dep = dep
diff --git a/meson_options.txt b/meson_options.txt
index 5cafe8b..3ff94dc 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -7,7 +7,7 @@ option('cogl', type : 'feature', value : 'disabled')
 option('directfb', type : 'feature', value : 'disabled')
 option('gl-backend', type : 'combo', value : 'disabled',
        # FIXME: https://github.com/mesonbuild/meson/issues/4566
-       choices : ['auto', 'gl', 'glesv2', 'glesv3', 'disabled'])
+       choices : ['auto', 'gl', 'glesv2', 'glesv3', 'angle', 'disabled'])
 option('glesv2', type : 'feature', value : 'disabled')
 option('glesv3', type : 'feature', value : 'disabled')
 option('drm', type : 'feature', value : 'disabled')
