diff --git a/meson.build b/meson.build
index 285d049..188da5d 100644
--- a/meson.build
+++ b/meson.build
@@ -889,7 +889,7 @@ deps += internal_deps
 subdir('src')
 
 incboilerplate = include_directories('boilerplate')
-if feature_conf.get('CAIRO_HAS_PNG_FUNCTIONS', 0) == 1
+if feature_conf.get('CAIRO_HAS_PNG_FUNCTIONS', 0) == 1 and not get_option('boilerplate').disabled()
   subdir('boilerplate')
 else
   libcairoboilerplate = []
diff --git a/meson_options.txt b/meson_options.txt
index a4d3dbd..9921174 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -23,6 +23,7 @@ option('xlib', type : 'feature', value : 'auto')
 option('zlib', type : 'feature', value : 'auto') # script, ps, pdf, xml surfaces
 
 # Tests
+option('boilerplate', type : 'feature', value : 'auto')
 option('tests', type : 'feature', value : 'auto')
 
 # Util deps
