diff --git a/meson.build b/meson.build
index 58ed0e3..1e2484b 100644
--- a/meson.build
+++ b/meson.build
@@ -485,7 +485,16 @@ if host_machine.system() == 'darwin' and not get_option('quartz').disabled()
   endif
 endif
 
-if host_machine.system() == 'windows'
+# When compiling cairo dll for 'windows' without 'win32' stuff (such as when compiling for UWP),
+# win32 mutex initialization is still required (see 'cairo-win32-system.c').
+if host_machine.system() == 'windows' and get_option('win32').disabled()
+  built_features += [
+    {
+      'name': 'cairo-win32-system-only',
+      'description': 'Microsoft Windows library initialization',
+    }
+  ]
+elif host_machine.system() == 'windows'
   win32_extra_deps = [
     cc.find_library('gdi32'),
     cc.find_library('msimg32'),
diff --git a/meson_options.txt b/meson_options.txt
index f55c3d5..1e74339 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -11,6 +11,7 @@ option('glesv3', type : 'feature', value : 'disabled')
 option('png', type : 'feature', value : 'auto') # png and svg surfaces
 option('quartz', type : 'feature', value : 'auto')
 option('tee', type : 'feature', value : 'disabled')
+option('win32', type : 'feature', value : 'auto')
 option('xcb', type : 'feature', value : 'auto')
 option('xlib', type : 'feature', value : 'auto')
 option('xlib-xcb', type : 'feature', value : 'disabled')
diff --git a/src/meson.build b/src/meson.build
index 612a3fc..50ba6da 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -181,6 +181,9 @@ cairo_feature_sources = {
     'win32/cairo-dwrite-font-public.c',
     'win32/cairo-dwrite-font.cpp',
   ],
+  'cairo-win32-system-only': [
+    'win32/cairo-win32-system.c',
+  ],
   'cairo-gl': [
     'cairo-gl-composite.c',
     'cairo-gl-device.c',
