diff --git a/meson.build b/meson.build
index 2609dc0..886296c 100644
--- a/meson.build
+++ b/meson.build
@@ -498,7 +498,16 @@ if host_machine.system() == 'darwin' and not get_option('quartz').disabled()
   endif
 endif
 
-if host_machine.system() == 'windows'
+# When compiling cairo dll for 'windows' without 'win32' stuff (such as when compiling for UWP),
+# win32 mutex initialization is still required (see 'cairo-win32-system.c').
+if host_machine.system() == 'windows' and get_option('win32').disabled()
+  built_features += [
+    {
+      'name': 'cairo-win32-system-only',
+      'description': 'Microsoft Windows library initialization',
+    }
+  ]
+elif host_machine.system() == 'windows'
   add_project_arguments('-DWIN32_LEAN_AND_MEAN', '-DNOMINMAX', language: ['c', 'cpp'])
 
   win32_extra_deps = [
diff --git a/meson_options.txt b/meson_options.txt
index 4bdfe8a..8d2e79e 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -7,6 +7,7 @@ option('freetype', type : 'feature', value : 'auto')
 option('png', type : 'feature', value : 'auto') # png and svg surfaces
 option('quartz', type : 'feature', value : 'auto')
 option('tee', type : 'feature', value : 'disabled')
+option('win32', type : 'feature', value : 'auto')
 option('xcb', type : 'feature', value : 'auto')
 option('xlib', type : 'feature', value : 'auto')
 option('xlib-xcb', type : 'feature', value : 'disabled')
diff --git a/src/meson.build b/src/meson.build
index 6866512..a75b0fc 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -182,6 +182,9 @@ cairo_feature_sources = {
   'cairo-dwrite-font': [
     'win32/cairo-dwrite-font.cpp',
   ],
+  'cairo-win32-system-only': [
+    'win32/cairo-win32-system.c',
+  ],
   'cairo-script': [
     'cairo-script-surface.c',
   ],
