diff --git a/meson.build b/meson.build
index 204bee2..285d049 100644
--- a/meson.build
+++ b/meson.build
@@ -382,7 +382,16 @@ if host_machine.system() == 'darwin' and not get_option('quartz').disabled()
   endif
 endif
 
-if host_machine.system() == 'windows'
+# When compiling cairo dll for 'windows' without 'win32' stuff (such as when compiling for UWP),
+# win32 mutex initialization is still required (see 'cairo-win32-system.c').
+if host_machine.system() == 'windows' and get_option('win32').disabled()
+  built_features += [
+    {
+      'name': 'cairo-win32-system-only',
+      'description': 'Microsoft Windows library initialization',
+    }
+  ]
+elif host_machine.system() == 'windows'
   win32_extra_deps = [
     cc.find_library('gdi32'),
     cc.find_library('msimg32'),
diff --git a/meson_options.txt b/meson_options.txt
index 3ff94dc..a4d3dbd 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -16,6 +16,7 @@ option('png', type : 'feature', value : 'auto') # png and svg surfaces
 option('quartz', type : 'feature', value : 'auto')
 option('qt', type : 'feature', value : 'disabled')
 option('tee', type : 'feature', value : 'disabled')
+option('win32', type : 'feature', value : 'auto')
 option('xcb', type : 'feature', value : 'auto')
 option('xlib', type : 'feature', value : 'auto')
 #option('xml', type : 'feature', value : 'disabled')
diff --git a/src/meson.build b/src/meson.build
index 5bd1a57..430bab0 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -179,6 +179,9 @@ cairo_feature_sources = {
   'cairo-win32-font': [
     'win32/cairo-win32-font.c',
   ],
+  'cairo-win32-system-only': [
+    'win32/cairo-win32-system.c',
+  ],
   'cairo-drm': [
     'drm/cairo-drm.c',
     'drm/cairo-drm-bo.c',
