diff --git a/meson.build b/meson.build
index cc13180..3fb6d38 100644
--- a/meson.build
+++ b/meson.build
@@ -484,7 +484,25 @@ if host_machine.system() == 'darwin' and not get_option('quartz').disabled()
   endif
 endif
 
-if host_machine.system() == 'windows'
+# When compiling cairo dll for 'windows' or 'UWP' without 'win32' stuff,
+# keep 'cairo-win32-system.c' code, just to be sure that Win32 mutexes are correctly initialized.
+if host_machine.system() == 'windows' and get_option('win32').disabled()
+  add_languages('cpp', native: false)
+
+  add_project_arguments('-DWIN32_LEAN_AND_MEAN', '-DNOMINMAX', language: ['c', 'cpp'])
+  if cc.get_define('WINAPI_FAMILY') == 'WINAPI_FAMILY_APP' or cc.get_define('WINAPI_FAMILY') == 'WINAPI_FAMILY_PC_APP' or cc.get_define('WINAPI_FAMILY') == '2'
+    add_project_arguments('-DWINVER=_WIN32_WINNT_WIN10', '-D_WIN32_WINNT=_WIN32_WINNT_WIN10', '-DNTDDI_VERSION=NTDDI_WIN10_RS3', language: ['c', 'cpp'])
+  else
+    add_project_arguments('-DWINVER=_WIN32_WINNT_WIN2K', '-D_WIN32_WINNT=_WIN32_WINNT_WIN2K', language: ['c', 'cpp'])
+  endif
+
+  built_features += [
+    {
+      'name': 'cairo-win32-system-only',
+      'description': 'Microsoft Windows library initialization',
+    }
+  ]
+elif host_machine.system() == 'windows'
   add_languages('cpp', native: false)
 
   add_project_arguments('-DWIN32_LEAN_AND_MEAN', '-DNOMINMAX', language: ['c', 'cpp'])
diff --git a/meson.options b/meson.options
index 8aa6814..92b7c9f 100644
--- a/meson.options
+++ b/meson.options
@@ -7,6 +7,7 @@ option('freetype', type : 'feature', value : 'auto')
 option('png', type : 'feature', value : 'auto') # png and svg surfaces
 option('quartz', type : 'feature', value : 'auto')
 option('tee', type : 'feature', value : 'auto')
+option('win32', type : 'feature', value : 'auto')
 option('xcb', type : 'feature', value : 'auto')
 option('xlib', type : 'feature', value : 'auto')
 option('xlib-xcb', type : 'feature', value : 'disabled')
diff --git a/src/meson.build b/src/meson.build
index ac06ac6..15f9ca6 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -182,6 +182,9 @@ cairo_feature_sources = {
   'cairo-dwrite-font': [
     'win32/cairo-dwrite-font.cpp',
   ],
+  'cairo-win32-system-only': [
+    'win32/cairo-win32-system.c',
+  ],
   'cairo-script': [
     'cairo-script-surface.c',
   ],
