diff --git a/portfile.cmake b/portfile.cmake
index a91bcce..d1ffa57 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -39,6 +39,19 @@ vcpkg_download_distfile(installer_path
     SHA512 "${sha}"
 )
 
+#my-change - begin
+# libiomp5md.lib always requires libiomp5md.dll presence, when crt-linkage is dynamic, regardless of vcpkg-library-linkage mode.
+# See https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html for linking commandline.
+# See https://www.intel.com/content/www/us/en/docs/onemkl/developer-guide-windows/2025-2/linking-with-compiler-run-time-libraries.html for /MT and /MD differences.
+#	/MT for linking with static Intel® oneAPI Math Kernel Library (oneMKL) libraries
+#	/MD for linking with dynamic Intel® oneAPI Math Kernel Library (oneMKL) libraries
+if(VCPKG_TARGET_IS_WINDOWS AND (VCPKG_CRT_LINKAGE STREQUAL "dynamic"))
+    set(MKL_LIBRARY_LINKAGE "dynamic")
+else()
+	set(MKL_LIBRARY_LINKAGE ${VCPKG_LIBRARY_LINKAGE})
+endif()
+#my-change - end
+
 # Note: intel_thread and lp64 are the defaults.
 set(interface "ilp64") # or ilp64; ilp == 64 bit int api
 #https://www.intel.com/content/www/us/en/develop/documentation/onemkl-linux-developer-guide/top/linking-your-application-with-onemkl/linking-in-detail/linking-with-interface-libraries/using-the-ilp64-interface-vs-lp64-interface.html
@@ -52,7 +65,7 @@ if(threading STREQUAL "intel_thread")
 else()
     string(SUBSTRING "${threading}" "0" "3" short_thread)
 endif()
-set(main_pc_file "mkl-${VCPKG_LIBRARY_LINKAGE}-${interface}-${short_thread}.pc")
+set(main_pc_file "mkl-${MKL_LIBRARY_LINKAGE}-${interface}-${short_thread}.pc")	#my-change
 
 # First extraction level: packages (from offline installer)
 set(extract_0_dir "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-extract")
@@ -128,7 +141,7 @@ vcpkg_execute_required_process(
 set(mkl_dir "${extract_1_dir}/_installdir/mkl/${mkl_short_version}")
 file(COPY "${mkl_dir}/include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
 file(COPY "${mkl_dir}/${package_libdir}/" DESTINATION "${CURRENT_PACKAGES_DIR}/lib/")
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+if(${MKL_LIBRARY_LINKAGE} STREQUAL "dynamic")	#my-change
   file(COPY "${mkl_dir}/${runtime_dir}/" DESTINATION "${CURRENT_PACKAGES_DIR}/bin/")
 endif()
 
@@ -138,14 +151,14 @@ vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/${main_pc_file}" "\$
 set(compiler_dir "${extract_1_dir}/_installdir/compiler/${mkl_short_version}")
 if(threading STREQUAL "intel_thread")
   file(COPY "${compiler_dir}/lib/" DESTINATION "${CURRENT_PACKAGES_DIR}/lib/")
-  if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+  if(${MKL_LIBRARY_LINKAGE} STREQUAL "dynamic")	#my-change
     file(COPY "${compiler_dir}/bin/" DESTINATION "${CURRENT_PACKAGES_DIR}/bin/")
   endif()
   file(COPY_FILE "${compiler_dir}/lib/pkgconfig/openmp.pc" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/libiomp5.pc")
   vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/${main_pc_file}" "openmp" "libiomp5")
 endif()
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+if(${MKL_LIBRARY_LINKAGE} STREQUAL "dynamic")	#my-change
     set(to_remove_suffix .a)
 else()
   if(VCPKG_TARGET_IS_WINDOWS)
@@ -177,7 +190,7 @@ endif()
 file(COPY "${mkl_dir}/lib/cmake/" DESTINATION "${CURRENT_PACKAGES_DIR}/share/")
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/mkl/MKLConfig.cmake" "MKL_CMAKE_PATH}/../../../" "MKL_CMAKE_PATH}/../../")
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/mkl/MKLConfig.cmake" "redist/\${MKL_ARCH}" "bin")
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+if(${MKL_LIBRARY_LINKAGE} STREQUAL "static")	#my-change
   vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/mkl/MKLConfig.cmake" "set(DEFAULT_MKL_LINK dynamic)" "set(DEFAULT_MKL_LINK static)")
   vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/mkl/MKLConfig.cmake" "set(LIB_EXT \".so\")" "set(LIB_EXT \".a\")")
 endif()
@@ -192,6 +205,7 @@ file(REMOVE_RECURSE
     "${extract_0_dir}"
     "${extract_1_dir}"
     "${CURRENT_PACKAGES_DIR}/bin/compiler"
+    "${CURRENT_PACKAGES_DIR}/lib/clang"	#my-change
     "${CURRENT_PACKAGES_DIR}/lib/cmake"
 )
 
