diff --git a/portfile.cmake b/portfile.cmake
index d077ee0..a859ff4 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -103,7 +103,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
     set(mkl_dir "${extract_1_dir}/Intel/Compiler/12.0/mkl/2023.0.0")
     file(COPY "${mkl_dir}/include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
     # see https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html for linking
-    if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    if((VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic") OR (VCPKG_CRT_LINKAGE STREQUAL "dynamic")) #my-change
       set(files "mkl_core_dll.lib" "mkl_${threading}_dll.lib" "mkl_intel_${interface}_dll.lib" "mkl_blas95_${interface}.lib" "mkl_lapack95_${interface}.lib") # "mkl_rt.lib" single dynamic lib with dynamic dispatch
       file(COPY "${mkl_dir}/redist/intel64/" DESTINATION "${CURRENT_PACKAGES_DIR}/bin") # Could probably be reduced instead of copying all
       if(NOT VCPKG_BUILD_TYPE)
