diff --git a/portfile.cmake b/portfile.cmake
index 0490c85..b223b65 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -15,6 +15,9 @@ vcpkg_from_github(
         protobuf.diff
 )
 
+#my change: enable pkgconfig for Windows.
+vcpkg_replace_string("${SOURCE_PATH}/CMakeLists.txt" "if (NOT MSVC)" "if (true)")
+
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SPM_ENABLE_SHARED)
 
 vcpkg_cmake_configure(
