diff --git a/portfile.cmake b/portfile.cmake
index c622dfd..1735cd3 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -1,3 +1,11 @@
+#my-change begin
+if ("ms-icu" IN_LIST FEATURES)
+  set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+  configure_file("${CURRENT_PACKAGES_DIR}/../ms-icu_${TARGET_TRIPLET}/share/ms-icu/copyright" "${CURRENT_PACKAGES_DIR}/share/icu/copyright" COPYONLY)
+  return()
+endif()
+#my-change end
+
 string(REGEX MATCH "^[0-9]*" ICU_VERSION_MAJOR "${VERSION}")
 string(REPLACE "." "_" VERSION2 "${VERSION}")
 string(REPLACE "." "-" VERSION3 "${VERSION}")
@@ -21,6 +29,13 @@ vcpkg_extract_source_archive(SOURCE_PATH
         fix-win-build.patch
 )
 
+#my-change begin: fix 'File name too long' error during install-icu phase, while executing 'ln -s' command.
+if(CMAKE_HOST_WIN32)
+    #Both winsymlinks and winsymlinks:nativestrict work fine, but winsymlinks:nativestrict may require Windows Developer Mode or administrative privileges.
+	set(ENV{MSYS} "winsymlinks")
+endif()
+#my-change end
+
 vcpkg_find_acquire_program(PYTHON3)
 set(ENV{PYTHON} "${PYTHON3}")
 
diff --git a/vcpkg.json b/vcpkg.json
index 88a574b..73ea31e 100644
--- a/vcpkg.json
+++ b/vcpkg.json
@@ -4,19 +4,38 @@
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
   "license": "ICU",
-  "supports": "!uwp",
   "dependencies": [
     {
       "name": "icu",
       "host": true,
       "features": [
         "tools"
-      ]
+      ],
+	  "platform": "!uwp"
+    },
+	{
+      "name": "icu",
+	  "features": [
+	    "ms-icu"
+	  ],
+	  "platform": "uwp"
     }
   ],
   "features": {
     "tools": {
       "description": "Build tools"
+    },
+	"ms-icu": {
+      "description": "Use MS-ICU on Windows",
+	  "supports": "windows",
+	  "dependencies": [
+	    {
+          "name": "ms-icu",
+          "features": [
+            "dummyheaders"
+          ]
+        }
+	  ]
     }
   }
 }
