diff --git a/portfile.cmake b/portfile.cmake
index 0beaa23..08e2c06 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -1,3 +1,11 @@
+#my-change begin
+if ("ms-icu" IN_LIST FEATURES)
+  set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+  configure_file("${CURRENT_PACKAGES_DIR}/../ms-icu_${TARGET_TRIPLET}/share/ms-icu/copyright" "${CURRENT_PACKAGES_DIR}/share/icu/copyright" COPYONLY)
+  return()
+endif()
+#my-change end
+
 string(REGEX MATCH "^[0-9]*" ICU_VERSION_MAJOR "${VERSION}")
 string(REPLACE "." "_" VERSION2 "${VERSION}")
 string(REPLACE "." "-" VERSION3 "${VERSION}")
@@ -23,6 +31,13 @@ vcpkg_extract_source_archive(SOURCE_PATH
         darwin-rpath.patch
 )
 
+#my-change begin: fix 'File name too long' error during install-icu phase, while executing 'ln -s' command.
+if(CMAKE_HOST_WIN32)
+    #Both winsymlinks and winsymlinks:nativestrict work fine, but winsymlinks:nativestrict may require Windows Developer Mode or administrative privileges.
+	set(ENV{MSYS} "winsymlinks")
+endif()
+#my-change end
+
 vcpkg_find_acquire_program(PYTHON3)
 set(ENV{PYTHON} "${PYTHON3}")
 
diff --git a/vcpkg.json b/vcpkg.json
index 441d2be..1a0dade 100644
--- a/vcpkg.json
+++ b/vcpkg.json
@@ -17,6 +17,18 @@
     "tools": {
       "description": "Build tools",
       "supports": "!uwp"
+    },
+	"ms-icu": {
+      "description": "Use MS-ICU on Windows",
+	  "supports": "windows",
+	  "dependencies": [
+	    {
+          "name": "ms-icu",
+          "features": [
+            "dummyheaders"
+          ]
+        }
+	  ]
     }
   }
 }
