diff --git a/portfile.cmake b/portfile.cmake
index d4fbbe0..1a1da46 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -137,6 +137,17 @@ if(threading STREQUAL "intel_thread")
   file(COPY "${compiler_dir}/lib/" DESTINATION "${CURRENT_PACKAGES_DIR}/lib/")
   file(COPY_FILE "${compiler_dir}/lib/pkgconfig/openmp.pc" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/libiomp5.pc")
   vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/${main_pc_file}" "openmp" "libiomp5")
+  # my-change - begin
+  # libiomp5md.lib always requires libiomp5md.dll presence.
+  # See https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html for linking commandline.
+  if (VCPKG_TARGET_IS_WINDOWS)
+    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/bin")
+    file(GLOB libiomp_dlls RELATIVE "${compiler_dir}/bin" "${compiler_dir}/bin/libiomp*.dll")
+	foreach(file IN LISTS libiomp_dlls)
+      file(COPY_FILE "${compiler_dir}/bin/${file}" "${CURRENT_PACKAGES_DIR}/bin/${file}")
+    endforeach()
+  endif()
+  # my-change - end
 endif()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
@@ -153,6 +164,16 @@ file(GLOB_RECURSE files_to_remove
     "${CURRENT_PACKAGES_DIR}/lib/*${to_remove_suffix}"
     "${CURRENT_PACKAGES_DIR}/lib/*${to_remove_suffix}.?"
 )
+
+#my-change - begin
+# libiomp5md.lib always requires libiomp5md.dll presence.
+# See https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html for linking commandline.
+if(VCPKG_TARGET_IS_WINDOWS AND (threading STREQUAL "intel_thread"))
+    list(FILTER files_to_remove EXCLUDE REGEX ".*libiomp.*\\.dll$")
+	#list(FILTER files_to_remove EXCLUDE REGEX ".*omptarget.*\\.dll$")
+endif()
+#my-change - end
+
 file(REMOVE ${files_to_remove})
 
 file(COPY_FILE "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/${main_pc_file}" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/mkl.pc")
