diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ecf285..209ff48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,31 +19,19 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif(NOT CMAKE_BUILD_TYPE)
 
-option(SPM_ENABLE_SHARED "" OFF)
-option(SPM_ENABLE_TCMALLOC "" OFF)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third_party/sentencepiece EXCLUDE_FROM_ALL)
-
-if(MSVC)
-  set(SENTENCEPIECE_CXX_FLAGS "/w")
-else()
-  set(SENTENCEPIECE_CXX_FLAGS "-Wno-all -Wno-extra -Wno-error")
-  if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    set(SENTENCEPIECE_CXX_FLAGS "${SENTENCEPIECE_CXX_FLAGS} -Wno-stringop-overflow")
-  endif()
-endif()
-
-set_target_properties(
-  sentencepiece-static
-  sentencepiece_train-static
-  PROPERTIES COMPILE_FLAGS ${SENTENCEPIECE_CXX_FLAGS}
-  )
+#sentencepiece begin
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(SENTENCEPIECE REQUIRED sentencepiece)
+#explicitly get abseil 'flags' dependencies, not referenced by sentencepiece pkgconfig files.
+pkg_check_modules(ABSL_FLAGS REQUIRED absl_flags)
+#sentencepiece end
 
 if(MSVC)
   if(NOT BUILD_SHARED_LIBS)
     if(CMAKE_VERSION VERSION_LESS "3.15.0")
       message(FATAL_ERROR "Use CMake 3.15 or later when setting BUILD_SHARED_LIBS to OFF")
     endif()
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+    #set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
   endif()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /wd4251 /d2FH4-")
   set(ICU_COMPONENTS uc dt)
@@ -63,7 +51,7 @@ set(INCLUDE_DIRECTORIES
 
   PRIVATE
   ${ICU_INCLUDE_DIRS}
-  ${CMAKE_CURRENT_SOURCE_DIR}/third_party/sentencepiece/src
+  ${SENTENCEPIECE_INCLUDE_DIRS}
   )
 
 set(PUBLIC_HEADERS
@@ -98,8 +86,8 @@ set(SOURCES
 
 list(APPEND LINK_LIBRARIES
   ${ICU_LIBRARIES}
-  sentencepiece-static
-  sentencepiece_train-static
+  ${SENTENCEPIECE_LINK_LIBRARIES}
+  ${ABSL_FLAGS_LINK_LIBRARIES}
   )
 
 if(COMMAND create_library)
